src/sentry/eventstream/snuba.py

```diff
@@ -78,31 +93,31 @@ class SnubaProtocolEventStream(EventStream):
 
     def _get_headers_for_insert(
         self,
-        event,
-        is_new,
-        is_regression,
-        is_new_group_environment,
-        primary_hash,
-        received_timestamp,  # type: float
-        skip_consume,
+        event: BaseEvent,
+        is_new: bool,
+        is_regression: bool,
+        is_new_group_environment: bool,
+        primary_hash: Optional[str],
```

Comment:
the neat thing about `__future__.annotations` is you can use `str | None` instead of `Optional[str]` and `str | int` instead of `Union[str, int]` -- `pyupgrade` would do this for us but we have that option turned off at the moment for reasons I don't know

Reply:
Optional is currently the dominant style, i am not sure if we want to migrate slowly or all at once

Reply:
Leaving it as-is. Let's fix it with `pyupgrade` later when we decide to move to this style.